engine/frontend/src/api/enclave.js

```diff
@@ -15,50 +26,61 @@ const createApiPromiseClient = (apiClient) => {
-    const response = respFromGrpc.toObject()
-    
-    // processing the data so that frontend can consume it! 
-    const responseProcessed = response.enclaveInfoMap.map(enclave => {
+    const respFromGrpc = await makeRestApiRequest(
+         "engine_api.EngineService/GetEnclaves",
+        {"field":""},
+        {"headers":{'Content-Type': "application/json"}}
+    )
+
+    const {data} = respFromGrpc
+    return Object.keys(data.enclaveInfo).map(key => {
+        const enclave = data.enclaveInfo[key]
         return {
-            uuid: enclave[0],
-            name: enclave[1].name,
-            created: enclave[1].creationTime.seconds,
-            status: enclave[1].apiContainerStatus,
-            apiClient: createApiPromiseClient(enclave[1].apiContainerHostMachineInfo) 
+            uuid: enclave.enclaveUuid,
+            name: enclave.name,
+            created: enclave.creationTime,
+            status: enclave.apiContainerStatus,
+            apiClient: createApiPromiseClient(enclave.apiContainerHostMachineInfo)
         }
-    });
+    })
 
-    return responseProcessed
 }
 
-export const getServiceLogs = async (enclaveName, serviceUuid) => {
-    const args = new GetServiceLogsArgs();
-
-    args.setEnclaveIdentifier(enclaveName);
-    const serviceUuidMapSet = args.getServiceUuidSetMap();
-    const isServiceUuidInSet = true;
-    serviceUuidMapSet.set(serviceUuid, isServiceUuidInSet)
-    args.setFollowLogs(true);
+export const createEnclave = async () => {
+    const data = {
+        apiContainerVersionTag: "",
+        apiContainerLogLevel: "info",
+        isPartitioningEnabled: false,
+    }
+    const response = await makeRestApiRequest("engine_api.EngineService/CreateEnclave", JSON.stringify(data), {"headers":{'Content-Type': "application/json"}})
 
-    const stream = engineClient.getServiceLogs(args, {});
-    return stream;
-}
+    const enclave = response.data.enclaveInfo;
+    const apiClient = createApiPromiseClient(enclave.apiContainerHostMachineInfo);
 
-export const createEnclave = async () => {
-    const enclaveArgs = new CreateEnclaveArgs();
-    enclaveArgs.setApiContainerVersionTag("")
-    enclaveArgs.setApiContainerLogLevel("info");
-    const enclaveGRPC = await engineClient.createEnclave(enclaveArgs, null)
-    const enclave = enclaveGRPC.toObject().enclaveInfo;
     return {
-        uuid: enclave.uuid,
+        uuid: enclave.enclaveUuid,
         name: enclave.name,
-        created: enclave.creationTime.seconds,
+        created: enclave.creationTime,
         status: enclave.apiContainerStatus,
-        apiClient: createApiPromiseClient(enclave.apiContainerHostMachineInfo) 
+        apiClient
+    }
+}
+
+export const getServiceLogs = async (enclaveName, serviceUuid) => {
+    const args = {
+        "enclaveIdentifier": enclaveName,
+        "serviceUuidSet": {
+            [serviceUuid]: true
+        },
+        followLogs: false,
```

Comment:
@adschwartz ideally we want this to be true; but seeing some weird behaviours where stream requests get stuck on pending;; however having it as false works fine - we get the logs but don;t tail it.. in the interest of time, i can look into that in the next pr ( known bug) .. while this can get merged for e2e enclave-manager. My hunch is that it's happening because I am not correctly cancelling the requests when I unmount the stream from react page; but will have better idea tomorrow hopefully.

Reply:
@Peeeekay agreed, good idea to look into if we are following the right procedure when handling the streams. 

If you need another day to merge this, then that can work too. 

Just for my curiosity, what does `followLogs=true` do?

Reply:
It allows us to tail logs and see the new logs  in real-time on the UI instead of refreshing the page

Reply:
it works now! used AbortController.